define/types.go

```diff
@@ -100,9 +101,40 @@ type Secret struct {
 
 // BuildOutputOptions contains the the outcome of parsing the value of a build --output flag
 type BuildOutputOption struct {
-	Path     string // Only valid if !IsStdout
-	IsDir    bool
+	ImageRef imageTypes.ImageReference
+	Image    string
 	IsStdout bool
+	Path     string // Only valid if !IsStdout
+	Push     bool
+	Type     BuildOutputType
+
+	// Deprecated: Use Type instead to determine output type
+	IsDir bool
```

Comment:
AFAICS `IsDir` is now never set. If so, keeping the field around is just as incompatible an API break as removing the field… and we might just as well remove it.

Alternatively, if we need to keep it, it must be set in a compatible way (and I don’t know what “a compatible way” would be for `Type == …Image`.)

Or, hypothetically, we could leave the `BuildOutputOption` and the existing code as is, parsing only local file outputs but keeping API compatibility, and introduce a new type and a new set of utilities that can represent image outputs. 

Buildah maintainers need to decide which one of the three it is; I don’t know. (My guess is the first one, at least I found no Podman user of `BuildOutputOption`, but I really lack the context.)

Reply:
> keeping the field around is just as incompatible an API break as removing the field… and we might just as well remove it.

From what I can tell, and I have very little context, the idea was to minimize disruption. Outright removing it might cause more disruption than keeping it around and not using it but I understand your concern with having dead code lying around.